File: msr/reference/typescript/caesar-cipher.ts
function shiftText(text: string, k: number): string {
  return [...text]
    .map((c) => {
      if (c < "a" || c > "z") return c;
      const off = (((c.charCodeAt(0) - 97 + k) % 26) + 26) % 26;
      return String.fromCharCode(97 + off);
    })
    .join("");
}

function encrypt(text: string, shift: number): string {
  return shiftText(text, shift);
}

function decrypt(text: string, shift: number): string {
  return shiftText(text, -shift);
}

File: msr/reference/rust/caesar-cipher.rs
fn shift_text(text: &str, k: i64) -> String {
    text.chars()
        .map(|c| {
            if c.is_ascii_lowercase() {
                let off = ((c as i64 - 97 + k).rem_euclid(26)) as u8;
                (b'a' + off) as char
            } else {
                c
            }
        })
        .collect()
}

fn encrypt(text: &str, shift: i64) -> String {
    shift_text(text, shift)
}

fn decrypt(text: &str, shift: i64) -> String {
    shift_text(text, -shift)
}
